From 33d0c45d390ce28e191350f003fced8b28a5f289 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sat, 20 Mar 2021 17:45:53 +0100
Subject: [PATCH] Explicitly build dbus-cppc-helper as static library

The project sets -fvisibility=hidden globally which causes linking
errors when CMake decides to build dbus-cppc-helper as shared library.

compiler_main.cpp:(.text.startup+0x1): undefined reference to `core::dbus::Compiler::main(int, char const**)'
---
 src/core/dbus/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/dbus/CMakeLists.txt b/src/core/dbus/CMakeLists.txt
index 4713d4d..441f98d 100644
--- a/src/core/dbus/CMakeLists.txt
+++ b/src/core/dbus/CMakeLists.txt
@@ -80,7 +80,7 @@ install(
 # thus collect symbols in this temporary helper library that is not
 # installed but just linked to the actual compiler executable.
 add_library(
-  dbus-cppc-helper
+  dbus-cppc-helper STATIC
 
   compiler.cpp
   generator.cpp
-- 
2.31.0

